atomic_reactor/plugins/cachi2_postprocess.py

```diff
@@ -141,6 +152,48 @@ def postprocess_git_submodules_global_sbom(self):
             json.dump(global_sbom_data, global_sbom_f)
             global_sbom_f.flush()
 
+    def get_remote_source_json_config(self, remote_source_path: str):
+        """Process injected remote source and returns"""
+
+        # Cachi2 is injecting config updates into app dir,
+        # we don't want them in source archive, however
+        # we need them for later rebuild and analysis, to mirror
+        # Cachito behavior.
+        # Cachi2 creates .build-config.json file that contains paths
+        # and templates how to modify files.
+        # Instead of replicating cachi2 templating logic, just use path
+        # and load already patched file.
+        # This is internal logic of cachi2, this may explode violently when
+        # format of .build-config.json changes. Make sure that cachi2 image
+        # is tested before promoted to prod.
+
+        json_config: List[Dict] = []
+        cachi2_build_config_path = os.path.join(
+            remote_source_path, CACHI2_BUILD_CONFIG_JSON)
+        if not os.path.exists(cachi2_build_config_path):
+            return json_config
+
+        with open(cachi2_build_config_path, 'r') as f:
+            cachi2_build_config = json.load(f)
+
+        injected_files = cachi2_build_config.get("project_files", [])
+        for injected_file in injected_files:
+            path = injected_file["abspath"]  # fail horribly if this is missing
```

Comment:
under what circumstances could this be missing? Do we want to fail with keyerror ?

Reply:
When format of cachi2 file changes

Reply:
yes